ouroboros-network/src/Ouroboros/Network/Protocol/ChainSync/Codec.hs

```diff
@@ -80,33 +85,56 @@ codecChainSync =
           return (SomeMessage MsgAwaitReply)
 
         (2, 3, ServerAgency (TokNext _)) -> do
-          h <- CBOR.decode
-          p <- CBOR.decode
+          h <- decodeHeader
+          p <- decodePoint
           return (SomeMessage (MsgRollForward h p))
 
         (3, 3, ServerAgency (TokNext _)) -> do
-          p1 <- CBOR.decode
-          p2 <- CBOR.decode
+          p1 <- decodePoint
+          p2 <- decodePoint
           return (SomeMessage (MsgRollBackward p1 p2))
 
         (4, 2, ClientAgency TokIdle) -> do
-          ps <- CBOR.decode
+          ps <- decodeList decodePoint
           return (SomeMessage (MsgFindIntersect ps))
 
         (5, 3, ServerAgency TokIntersect) -> do
-          p1 <- CBOR.decode
-          p2 <- CBOR.decode
+          p1 <- decodePoint
+          p2 <- decodePoint
           return (SomeMessage (MsgIntersectImproved p1 p2))
 
         (6, 2, ServerAgency TokIntersect) -> do
-          p <- CBOR.decode
+          p <- decodePoint
           return (SomeMessage (MsgIntersectUnchanged p))
 
         (7, 1, ClientAgency TokIdle) ->
           return (SomeMessage MsgDone)
 
         _ -> fail ("codecChainSync: unexpected key " ++ show (key, len))
 
+codecSerialise :: forall header point m.
+                  (MonadST m, Serialise header, Serialise point)
+               => Codec (ChainSync header point)
+                        CBOR.DeserialiseFailure m ByteString
+codecSerialise =
+  codecChainSync
+    Serialise.encode
+    Serialise.decode
+    Serialise.encode
+    Serialise.decode
+
+
+encodeList :: (a -> CBOR.Encoding) -> [a] -> CBOR.Encoding
+encodeList _   [] = CBOR.encodeListLen 0
+encodeList enc xs = CBOR.encodeListLenIndef
+                 <> Prelude.foldr (\x r -> enc x <> r) CBOR.encodeBreak xs
+
+decodeList :: CBOR.Decoder s a -> CBOR.Decoder s [a]
+decodeList dec = do
+  mn <- CBOR.decodeListLenOrIndef
+  case mn of
+    Nothing -> CBOR.decodeSequenceLenIndef (flip (:)) [] reverse   dec
+    Just n  -> CBOR.decodeSequenceLenN     (flip (:)) [] reverse n dec
```

Comment:
`encodeList` and `decodeList` copied from the `serialise` package. Unfortunately it does not export them directly, only via the `Serialise a => Serialise [a]` class.